src/coreclr/jit/gentree.cpp

```diff
@@ -30912,9 +30912,9 @@ GenTree* Compiler::gtFoldExprHWIntrinsic(GenTreeHWIntrinsic* tree)
 
                             int64_t shiftAmount = otherNode->AsVecCon()->GetElementIntegral(TYP_LONG, 0);
 
-                            if ((genTypeSize(simdBaseType) != 8) && (shiftAmount > INT_MAX))
+                            if ((uint64_t)shiftAmount >= ((uint64_t)genTypeSize(simdBaseType) * BITS_PER_BYTE))
```

Comment:
Does this one also need to be fixed in valuenum? It's also doing:
```
                        uint64_t shiftAmount = GetConstantSimd16(arg1VN).u64[0];

                        if (genTypeSize(baseType) != 8)
                        {
                            if (shiftAmount > INT_MAX)
                            {
                                // Ensure we don't lose track the the amount is an overshift
                                shiftAmount = -1;
                            }
                            arg1VN = VNForIntCon(static_cast<int32_t>(shiftAmount));
                        }
                        else
                        {
                            arg1VN = VNForLongCon(static_cast<int64_t>(shiftAmount));
                        }
                    }
```

Reply:
Ah good find! addressed